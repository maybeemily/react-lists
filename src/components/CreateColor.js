import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class CreateColor extends PureComponent {
  static propTypes = {
    addColor: PropTypes.func.isRequired
  };

  state = {
    name: '', //waiting for input
    color: '#000000'
  }

  //on submit, change the state to the values inputted
  handleSubmit = event => {
    event.preventDefault();
    const { name, color } = this.state;
    this.props.addColor({ name, hex: color });
  }

  //on change of color, change the state of the color

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { color, name } = this.state;
    return (
      <form onSubmit={this.handleSubmit}>
        <input name="name" type="text" value={name} onChange={this.handleChange}/>
        <input name="color" type="color" value={color} onChange={this.handleChange}/>
        <button>Add Color</button>
      </form>
    );
  }
}



//import things
//proptypes - add color function
//state
//handle submit with prevent default
//handle change
//render form
