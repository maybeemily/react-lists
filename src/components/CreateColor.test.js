import React from 'react';
import { shallow } from 'enzyme';
import CreateColor from './CreateColor';

describe('Create Color test', () => {
  it('renders Create Color', () => {
    const wrapper = shallow(<CreateColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
