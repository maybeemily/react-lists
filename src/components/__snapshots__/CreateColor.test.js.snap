// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Color test renders Create Color 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="#000000"
  />
  <button>
    Add Color
  </button>
</form>
`;
